0.6/Q1 0.6.txt

```diff
@@ -0,0 +1,167 @@
+Answer 1 
+
+i) Function filterByName that accepts a string as a parameter and returns an array with only those objects where the first_name field includes that string.
+
+function filterByName(str)
+    {
+        let ans = (arr.find(details => details.first_name==str))
+        return ans;
+    }
+console.log(filterByName("Sam"))
+
+Output :
+
+{id: 8, first_name: 'Sam', email: 'scorck7@sitemeter.com', date_of_birth: '2020/08/30'}
+date_of_birth: "2020/08/30"
+email: "scorck7@sitemeter.com"
+first_name: "Sam"
+id: 8
+
+find() only returns the first occurence but if there are multiple occurences then we can use filter to get all the values satisfying the condition.
+In other words the filter() method returns a new array with all elements that pass the test defined by the given function whereas ,
+the find() method returns the value of the first array element that satisfies the provided test function.
+
+function filterByName(str)
+    {
+        let ans = (arr.filter(details => details.first_name==str))
```

Comment:
first - Always try to use === when comparing
second here in the question its asked to find those objects whose firstname includes given string
So you should use include function details.first_name.includes(str)